core/gas.c

```diff
@@ -139,3 +140,32 @@ void fill_pressures(struct gas_pressures *pressures, const double amb_pressure,
 		}
 	}
 }
+
+enum gastype gasmix_to_type(struct gasmix mix)
+{
+	if (gasmix_is_air(mix))
+		return GASTYPE_AIR;
+	if (mix.o2.permille >= 980)
+		return GASTYPE_OXYGEN;
+	if (mix.he.permille == 0)
+		return mix.o2.permille >= 230 ? GASTYPE_NITROX : GASTYPE_AIR;
+	if (mix.o2.permille <= 180)
+		return GASTYPE_HYPOXIC_TRIMIX;
+	return mix.o2.permille <= 230 ? GASTYPE_NORMOXIC_TRIMIX : GASTYPE_HYPEROXIC_TRIMIX;
```

Comment:
I like it.
I'm not sure if there are people diving with hypoxic nitrox, TBH - you call that air here (and I hadn't mentioned it earlier). But again, I think this is fine.